src/kOS/Module/kOSProcessor.cs

```diff
@@ -628,12 +637,29 @@ private void ProcessElectricity(Part partObj, float time)
         {
             if (ProcessorMode == ProcessorModes.OFF) return;
 
-            RequiredPower = shared.VolumeMgr.CurrentRequiredPower;
-            var electricReq = time * RequiredPower;
-            var result = partObj.RequestResource("ElectricCharge", electricReq) / electricReq;
-
-            var newMode = (result < 0.5f) ? ProcessorModes.STARVED : ProcessorModes.READY;
-            SetMode(newMode);
+            double volumePower = 0;
+            var volume = shared.VolumeMgr.CurrentVolume;
+            if (volume.Name == "Archive")
+            {
+                // TODO: Like with the default value of ECPerBytePerSecond, this corresponds to the existing value.
+                // It should be adjusted for balance.
+                volumePower = 50000 * ECPerBytePerSecond;
+            }
+            else
+            {
+                volumePower = volume.Capacity * ECPerBytePerSecond;
+            }
+            RequiredPower = (float)(volumePower + SafeHouse.Config.InstructionsPerUpdate * ECPerInstruction / TimeWarp.fixedDeltaTime);
+            var request = volumePower * time + shared.Cpu.InstructionsThisUpdate * ECPerInstruction;
+            var available = partObj.RequestResource("ElectricCharge", request);
+            if (available / request > 0.5f)
```

Comment:
It looks like the logic is something along the lines of finding a way to take the time when power depletion will happen and "round it" up or down to the nearest FixedUpdate boundary.  If there's enough EC left for somewhere between 50% and 100% of a full physics tick, then "round up" to doing one more full physics tick.  If there's between 0% and 50% as much as you need for a full physics tick, then "round down" to zero more ticks.

But in our case I think this logic may not be totally appropriate as we have a varying drain per update.  The current update tick may have taken a lot more power drain than the next one will.  

But in the end it probably doesn't matter much.  Nobody's going to notice that the CPU got power starved half a second sooner than it should have because it treated a battery charge of 0.1% as if it was 0.0%.  Never mind my complaint.  Just leave it.